data/publiccloud/log_instance.sh

```diff
@@ -102,7 +104,7 @@ azure_start_log()
     az vm boot-diagnostics get-boot-log --ids "$INSTANCE_ID" > "${OUTPUT_DIR}/$CNT""_boot_log_start.txt" 2>&1
     set -e
     nohup ssh $SSH_OPTS "azureuser@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 &
-    echo "$!" >> "$PID_FILE"
+    echo $! >> "$PID_FILE"
```

Comment:
and here as well I'd revert it.

Reply:
OK, after you pointed out that `zsh` has some problems with `$!` (it puts `dquote>` on my system) feel free to remove quotes (i.e. use your original approach). I thought just echo to file would be a problem but it's not.